src/platform/bouffalolab/BL602/NetworkCommissioningDriver.cpp

```diff
@@ -157,12 +167,24 @@ CHIP_ERROR BLWiFiDriver::ConnectWiFiNetwork(const char * ssid, uint8_t ssidLen,
 
     char wifi_ssid[64] = { 0 };
     char passwd[64]    = { 0 };
-    // Set the wifi configuration
+    int state          = 0;
+
+    wifi_mgmr_sta_disconnect();
+    vTaskDelay(WIFI_STA_DISCONNECT_DELAY);
+
+    wifi_mgmr_sta_disable(NULL);
+    wifi_mgmr_state_get(&state);
+    while (state != WIFI_STATE_IDLE)
+    {
+        wifi_mgmr_state_get(&state);
+        vTaskDelay(100);
```

Comment:
what is the 100 here? why is it ok to use ticks which depend on tick rate instead of computing a standard time. At least a comment of why we do not care is appropriate.